packages/pytest-simcore/src/pytest_simcore/httpbin_service.py

```diff
@@ -0,0 +1,64 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+
+
+import logging
+from contextlib import suppress
+from typing import Iterable
+
+import aiohttp.test_utils
+import docker
+import pytest
+import requests
+import requests.exceptions
+from docker.errors import APIError
+from pydantic import HttpUrl, parse_obj_as
+from tenacity import retry
+from tenacity.after import after_log
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_fixed
+
+from .helpers.utils_docker import get_localhost_ip
+
+
+@pytest.fixture(scope="session")
+def httpbin_base_url() -> Iterable[HttpUrl]:
+    """
+    Implemented as a fixture since it cannot rely on full availability of https://httpbin.org/ during testing
+    """
+    ip_address = get_localhost_ip()
+    port = aiohttp.test_utils.unused_port()
+    base_url = f"http://{ip_address}:{port}"
+
+    client = docker.from_env()
+    container_name = "httpbin-fixture"
+    try:
+        container = client.containers.run(
+            image="kennethreitz/httpbin",
+            ports={80: port},
+            name=container_name,
+            detach=True,
+        )
+        print(container)
+
+        @retry(
+            wait=wait_fixed(1),
+            retry=retry_if_exception_type(requests.exceptions.RequestException),
+            stop=stop_after_delay(15),
```

Comment:
Just a tough. I don't know how fast this service usually starts, consider this running on low end CI machines, it needs to have enough time to start. If 15 second jus just enough consider doubling it to avoid flaky tests

Reply:
Noted. thx